samples/snapshot_operations.py

```diff
@@ -0,0 +1,232 @@
+
+    # Loop looking for updates till the state moves to a completed state.
+    taskName = task.GetInfo().GetName()
+    update = pc.WaitForUpdates(None)
+    state = task.GetInfo().GetState()
+    while state != vim.TaskInfo.State.success and \
+            state != vim.TaskInfo.State.error:
+        if (state == 'running') and (taskName.info.name != "Destroy"):
+            # check to see if VM needs to ask a question, thow exception
+            vm = task.GetInfo().GetEntity()
+            if vm is not None and isinstance(vm, vim.VirtualMachine):
+                qst = vm.GetRuntime().GetQuestion()
+            if qst is not None:
+                raise Exception("Task blocked, User Intervention required")
+
+    update = pc.WaitForUpdates(update.GetVersion())
+    state = task.GetInfo().GetState()
+
+    filter.Destroy()
+    if state == "error" and raiseOnError:
+        raise task.GetInfo().GetError()
+
+    return state
+
+
+def invoke_and_track(func, *args, **kw):
+    try:
+        task = func(*args, **kw)
+        wait_for_task(task)
+    except:
+        raise
+
+
+def list_snapshots_recursively(snapshots):
+    snapshot_data = []
+    snap_text = ""
+    for snapshot in snapshots:
+        snap_text = "Name: %s; Description: %s; CreateTime: %s; State: %s" % (
+                                        snapshot.name, snapshot.description,
+                                        snapshot.createTime, snapshot.state)
+        snapshot_data.append(snap_text)
+        snapshot_data = snapshot_data + list_snapshots_recursively(
+                                        snapshot.childSnapshotList)
+    return snapshot_data
+
+
+def get_snapshots_by_name_recursively(snapshots, snapname):
+    snap_obj = []
+    for snapshot in snapshots:
+        if snapshot.name == snapname:
+            snap_obj.append(snapshot)
+        else:
+            snap_obj = snap_obj + get_snapshots_by_name_recursively(
+                                    snapshot.childSnapshotList, snapname)
+    return snap_obj
+
+
+def get_current_snap_obj(snapshots, snapob):
+    snap_obj = []
+    for snapshot in snapshots:
+        if snapshot.snapshot == snapob:
+            snap_obj.append(snapshot)
+        snap_obj = snap_obj + get_current_snap_obj(
+                                snapshot.childSnapshotList, snapob)
+    return snap_obj
+
+
+def main():
+
+    try:
+        si = None
+        try:
+            print("Trying to connect to VCENTER SERVER . . .")
+
+            context = None
+            if inputs['ignore_ssl']:
+                context = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
+                context.verify_mode = ssl.CERT_NONE
+
+            si = connect.Connect(inputs['vcenter_ip'], 443,
+                                 inputs['vcenter_user'], inputs[
+                                     'vcenter_password'],
+                                 sslContext=context)
+        except IOError, e:
+            pass
+            atexit.register(Disconnect, si)
```

Comment:
This doens't look right to me. We only disconnect when there is an exception? Catching exception is not necessary here cause there is nothing to recover here.